doc/mdc-migration.md

```diff
@@ -0,0 +1,56 @@
+# Material Design Components (MDC) Migration
+## Status:
+Accepted on 2020-04-16
+
+## Context:
+We have been using `Theme.Design.*` theme from the start, while that worked perfectly it did not allow us to use new material components because those are dependent on `Theme.MaterialComponents.*` app theme. Using new material components allows us to have much more control over component theming like text styles, icon & icon gravity, strokes, corners, etc.,
+
```

Comment:
Trailing `,`. Please remove.
Also add a line saying that the following are the problems we are facing with our current theme.